src/pkg/packaging/windows/targetingpack/generatemsi.ps1

```diff
@@ -0,0 +1,162 @@
+
+$InstallFileswsx = "$WixObjRoot\install-files.wxs"
+$InstallFilesWixobj = "$WixObjRoot\install-files.wixobj"
+
+function RunHeat
+{
+    $result = $true
+    pushd "$WixRoot"
+
+    Write-Host Running heat.. to $InstallFileswsx
+    write-host "Root $TargetingPackPublishRoot"
+
+    .\heat.exe dir `"$TargetingPackPublishRoot`" `
+    -nologo `
+    -template fragment `
+    -sreg -gg `
+    -var var.TargetingPackSrc `
+    -cg InstallFiles `
+    -srd `
+    -dr DOTNETHOME `
+    -out $InstallFileswsx | Out-Host
+
+    if($LastExitCode -ne 0)
+    {
+        $result = $false
+        Write-Host "Heat failed with exit code $LastExitCode."
+    }
+
+    popd
+    return $result
+}
+
+function RunCandle
+{
+    $result = $true
+    pushd "$WixRoot"
+
+    Write-Host Running candle..
+    $AuthWsxRoot =  Join-Path $PackagingRoot "windows\targetingpack"
+
+    $ComponentVersion = $TargetingPackNugetVersion.Replace('-', '_');
+
+    .\candle.exe -nologo `
+        -out "$WixObjRoot\" `
+        -dTargetingPackSrc="$TargetingPackPublishRoot" `
+        -dMicrosoftEula="$PackagingRoot\windows\eula.rtf" `
+        -dProductMoniker="$ProductMoniker" `
+        -dBuildVersion="$TargetingPackMSIVersion" `
+        -dNugetVersion="$TargetingPackNugetVersion" `
+        -dComponentVersion="$ComponentVersion" `
+        -dTargetArchitecture="$TargetArchitecture" `
+        -dUpgradeCode="$TargetingPackUpgradeCode" `
+        -arch $Architecture `
+        -ext WixDependencyExtension.dll `
+        "$AuthWsxRoot\targetingpack.wxs" `
+        "$AuthWsxRoot\provider.wxs" `
+        $InstallFileswsx | Out-Host
+
+    if($LastExitCode -ne 0)
+    {
+        $result = $false
+        Write-Host "Candle failed with exit code $LastExitCode."
+    }
+
+    popd
+    return $result
+}
+
+function RunLight
+{
+    $result = $true
+    pushd "$WixRoot"
+
+    Write-Host Running light..
+
+    .\light.exe -nologo `
+        -ext WixUIExtension.dll `
+        -ext WixDependencyExtension.dll `
+        -ext WixUtilExtension.dll `
+        -cultures:en-us `
+        "$WixObjRoot\targetingpack.wixobj" `
+        "$WixObjRoot\provider.wixobj" `
+        "$InstallFilesWixobj" `
+        -out $TargetingPackMSIOutput | Out-Host
+
+    if($LastExitCode -ne 0)
+    {
+        $result = $false
+        Write-Host "Light failed with exit code $LastExitCode."
+    }
+
+    popd
+    return $result
+}
+
+if(!(Test-Path $TargetingPackPublishRoot))
+{
+    throw "$TargetingPackPublishRoot not found"
+}
+
+if(!(Test-Path $WixObjRoot))
+{
+    throw "$WixObjRoot not found"
+}
+
+Write-Host "Creating Targeting Pack MSI at $TargetingPackMSIOutput"
+
+if([string]::IsNullOrEmpty($WixRoot))
+{
+    Exit -1
+}
+
+if(-Not (RunHeat))
+{
+    Exit -1
+}
+
+if(-Not (RunCandle))
+{
+    Exit -1
+}
+
+if(-Not (RunLight))
+{
+    Exit -1
+}
+
+if(!(Test-Path $TargetingPackMSIOutput))
+{
+    throw "Unable to create the Targeting Pack MSI."
+    Exit -1
```

Comment:
Does this get executed?  Doesn't throw prevent Exit from getting executed?

Why is there a mixture of using Exit -1 vs throwing?  The script doesn't seem consistent.